### PR TITLE
 FIX: Force read the last collapsed or hidden post(s) in a topic

### PR DESCRIPTION
**Pull request by khalilovcmded**
## Why?

Aiming to fix [this bug](https://meta.discourse.org/t/will-discourse-ever-gain-ignore-user-functionality/1525/79).

**TL;DR:** If a new post (or few posts) are the last ones in a topic, and are collapsed / hidden, a User will always see the unread counter for this topic (since the posts are never marked as read/seen).
Discourse

## How?

The way it works is by trying to guesstimate if there is gap of more than 1 post between the **last post number** and the **highest seen number** (highest seen number is decided ONLY based on visible posts). 

## UI

1. You will notice I will have 2 unread posts (these are hidden/collapsed posts)

Discourse

2. When I go to the topic, you will NOT see any of the hidden/collapsed posts

What_is_the_best_time_to_hike_Mt__Fuji__⛰_-_Discourse

3. Then when I go back to the topic list view, you will see that the unread/unseen marker is gone

Discourse

### CODE REVIEW

**Comment by SamSaffron** [comment]
this feels very very hacky to me. 

Instead it should be calling `screenTrack.setIgnoredPosts(ignoredPostNumbers);` Then I guess it cant track 1ms for every ignored postNumber. 

It is a far more complex fix cause it runs deeper. But is far easier to comprehend.

**Comment by SamSaffron** [comment]
closing for now in favor of a different fix
